container_images/concourse-metrics/pkg/monitoring/requests_test.go

```diff
@@ -0,0 +1,96 @@
+	assertEqual(t, result.TimeSeries[0].Metric.Labels["result_state"], resultState)
+	assertEqual(t, result.TimeSeries[0].Resource.Type, "generic_task")
+	assertEqual(t, result.TimeSeries[0].Resource.Labels["project_id"], projectId)
+	assertEqual(t, result.TimeSeries[0].Resource.Labels["location"], zone)
+	assertEqual(t, result.TimeSeries[0].Resource.Labels["namespace"], pipeline)
+	assertEqual(t, result.TimeSeries[0].Resource.Labels["job"], job)
+	assertEqual(t, result.TimeSeries[0].Resource.Labels["task_id"], task)
+	assertEqual(t, result.TimeSeries[0].Points[0].Interval.EndTime.Seconds, validEndTimestamp/1000)
+	assertEqual(t, result.TimeSeries[0].Points[0].Value.GetInt64Value(), validEndTimestamp-startTimestamp)
+}
+
+func TestBuildJobResultRequestInputValidation(t *testing.T) {
+	// Instantiated since you can't make pointers to int literals (since they are not addressable).
+	validEndTimestamp := int64(1620363568000)
+
+	tests := []struct {
+		args    JobResultArgs
+		message string
+	}{
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: job, Pipeline: pipeline, ProjectId: projectId, ResultState: resultState, Task: task, Zone: " "}, "empty zone"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: job, Pipeline: pipeline, ProjectId: projectId, ResultState: resultState, Task: " ", Zone: zone}, "empty task"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: job, Pipeline: pipeline, ProjectId: projectId, ResultState: "other", Task: task, Zone: zone}, "invalid state"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: job, Pipeline: pipeline, ProjectId: " ", ResultState: resultState, Task: task, Zone: zone}, "empty project"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: job, Pipeline: " ", ProjectId: projectId, ResultState: resultState, Task: task, Zone: zone}, "empty pipeline"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: startTimestamp, Job: " ", Pipeline: pipeline, ProjectId: projectId, ResultState: resultState, Task: task, Zone: zone}, "empty job"},
+		{JobResultArgs{EndTimestamp: &validEndTimestamp, MetricPath: metricPath, StartTimestamp: validEndTimestamp + 1000, Job: job, Pipeline: pipeline, ProjectId: projectId, ResultState: resultState, Task: task, Zone: zone}, "end-timestamp cannot occur before start-timestamp"},
+	}
+	for _, test := range tests {
+		if _, err := BuildJobResultRequest(test.args); err == nil || !strings.Contains(err.Error(), test.message) {
+			t.Errorf("BuildJobResultRequest(%+v) = (_, %v); want error to contain %s", test.args, err, test.message)
+		}
+	}
+}
+
+func assertEqual(t *testing.T, x interface{}, y interface{}) {
```

Comment:
There's `github.com/stretchr/testify/assert` for asserts.

Reply:
Ah I didn't know this existed; thanks for the pointer. I'll swap assert usage out for testify in a separate PR.